src/core/Akka.Cluster/Cluster.cs

```diff
@@ -563,22 +564,38 @@ internal void Shutdown()
 
         private readonly IActorRef _clusterDaemons;
         private IActorRef _clusterCore;
+        private Task<IActorRef> _clusterCoreTask;
 
         /// <summary>
         /// TBD
         /// </summary>
+        [Obsolete("use TellCoreSafe()")]
         internal IActorRef ClusterCore
         {
             get
             {
-                if (_clusterCore == null)
+                if (_clusterCore is null)
                 {
-                    _clusterCore = GetClusterCoreRef().Result;
+                    if (_clusterCoreTask is null)
+                        throw new InvalidOperationException();
+
+                    _clusterCore = _clusterCoreTask.Result;
                 }
                 return _clusterCore;
             }
         }
 
+        /// <summary>
+        /// TBD
```

Comment:
Should add a real-comment here to explain what this does - as the old code did (i.e. "we have to wait for cluster core to startup before we can use it)

Reply:
done